kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/datatest/DataTest.kt

```diff
@@ -0,0 +1,123 @@
+package com.sksamuel.kotest.engine.datatest
+
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.DescriptionName
+import io.kotest.core.test.TestCase
+import io.kotest.core.test.TestResult
+import io.kotest.core.test.TestStatus
+import io.kotest.core.test.createTestName
+import io.kotest.engine.KotestEngineLauncher
+import io.kotest.engine.listener.TestEngineListener
+import io.kotest.matchers.shouldBe
+
+class DataTest : FunSpec() {
+   init {
+
+      test("free spec should support data tests") {
+         val results = mutableMapOf<DescriptionName.TestName, TestStatus>()
+         val listener = object : TestEngineListener {
```

Comment:
extract `TestEngineListener`, since it's repeated in all tests.

Reply:
Done